core/src/main/scala/shapeless/ops/records.scala

```diff
@@ -49,37 +50,35 @@ package record {
 
   }
 
+  //used for binary compatibility with 2.3.0
   class UnsafeSelector(i: Int) extends Selector[HList, Any] {
     type Out = Any
-    def apply(l: HList): Any = HList.unsafeGet(l, i)
+    def apply(l: HList): Any = HList.unsafeCrud(l, i, identity, false)._1
```

Comment:
Is this change necessary. I think the original `unsafeGet` will have the same compile time performance, so this change increases the complexity of the code slightly with no obvious gain.